src/mongo/editors/MongoCollectionNodeEditor.ts

```diff
@@ -38,6 +38,8 @@ export class MongoCollectionNodeEditor implements ICosmosEditor<IMongoDocument[]
             const documentNode = documentNodes.find((node) => node.treeItem.document._id.toString() === updatedDoc._id.toString());
             if (documentNode) {
                 documentNode.treeItem.document = updatedDoc;
+                documentNode.treeItem.label = getDocumentTreeItemLabel(documentNode.treeItem.document);
```

Comment:
I think the right thing to do here is:

1) put the logic to set _label in refreshLabel()
2) call documentNode.refresh() whenever you want the label to get refreshed